src/realm/collection_parent.hpp

```diff
@@ -72,8 +72,51 @@ enum class UpdateStatus {
     NoChange,
 };
 
+class ColIndex {
+public:
+    ColIndex()
+    {
+        value = uint32_t(-1);
+    }
+    explicit ColIndex(int64_t val)
+    {
+        value = uint32_t(val);
+    }
+    ColIndex(ColKey col_key, int64_t key)
+    {
+        value = uint32_t(col_key.get_index().val);
```

Comment:
Probably we can use a comment here about the composition of the value itself and what represents and why we need this structure, instead of just using the `ColKey`. It seems to me that:
`is_collection` -> Bit 32
`index_value` -> Bits [31...0] 
`key_value` -> Bits [15...0]

Reply:
Switched to using bit fields. I hope this is more readable